wzdx/raw_to_standard/icone.py

```diff
@@ -0,0 +1,209 @@
+        print(
+            f"Successfully generated standard message files: {generated_files_list}")
+    else:
+        logging.warning(
+            "Standard message generation failed. See messages printed above")
+
+
+def generate_standard_messages_from_string(input_file_contents):
+    raw_messages = generate_raw_messages(input_file_contents)
+    standard_messages = []
+    for message in raw_messages:
+        standard_messages.append(
+            generate_rtdh_standard_message_from_raw_single(message))
+    return standard_messages
+
+
+def generate_raw_messages(message, invalid_messages_callback=None):
+    response_xml = ET.fromstring(message)
+    msg_lst = response_xml.findall('incident')
+    messages = []
+
+    # Loop through all elements and print each element to PubSub
+    for msg in msg_lst:
+        print(msg)
+        incident = ET.tostring(msg, encoding='utf8')
+        obj = wzdx_translator.parse_xml_to_dict(incident)
+        if not validate_incident(obj.get('incident', {})):
+            if invalid_messages_callback:
+                invalid_messages_callback(obj)
+        else:
+            messages.append(incident)
+
+    return messages
+
+
+def generate_rtdh_standard_message_from_raw_single(raw_message_xml):
+    obj = wzdx_translator.parse_xml_to_dict(raw_message_xml)
+    pd = PathDict(obj)
+    standard_message = create_rtdh_standard_msg(pd)
+    return standard_message
+
+
+# parse script command line arguments
+def parse_rtdh_arguments():
+    parser = argparse.ArgumentParser(
+        description='Translate iCone data to RTDH Standard')
+    parser.add_argument('--version', action='version',
+                        version=f'{PROGRAM_NAME} {PROGRAM_VERSION}')
+    parser.add_argument('iconeFile', help='icone file path')
+    parser.add_argument('--outputDir', required=False,
+                        default='./', help='output directory')
+
+    args = parser.parse_args()
+    return args.iconeFile, args.outputDir
+
+
+# function to parse polyline to geometry line string
+# input: "37.1571990,-84.1128540,37.1686478,-84.1238971" (lat, long)
+# output: [[-84.1128540, 37.1571990], [-84.1238971, 37.1686478]] (long, lat)
+def parse_icone_polyline(polylinestring):
+    if not polylinestring or type(polylinestring) != str:
+        return None
+    # polyline rightnow is a list which has an empty string in it.
+    polyline = polylinestring.split(',')
+    coordinates = []
+    for i in range(0, len(polyline)-1, 2):
+        try:
+            coordinates.append([float(polyline[i + 1]), float(polyline[i])])
+        except ValueError as e:
+            logging.warning('failed to parse polyline!')
+            return []
+    return coordinates
+
+
+def get_sensor_list(incident):
+    devices = []
+    for key in ['sensor', 'radar', 'display', 'message', 'marker', 'status']:
+        obj = incident.get(f"{key}")
+        print("-------------------")
+        print(key)
+        print(obj)
+        print(type(obj))
```

Comment:
Do we want to leave these print statements in here? Seems like maybe just debug level.

Reply:
haha shoot, yeah that was totally just from debugging. removed